packages/introspect/src/tests/pg_types.test.ts

```diff
@@ -0,0 +1,288 @@
+import { describe, expect, test } from 'vitest';
+import { Pool, types } from 'pg';
+import Ajv2020 from 'ajv/dist/2020';
+import { generate } from '../generate';
+
+interface Column {
+    type: string;
+    examples: [string, string];
+}
+
+// Database connection configuration
+const client = new Pool({
+    connectionString: 'postgresql://postgres:postgres@localhost:5432/postgres',
+    max: 10,
+});
+
+// Use the OIDs imported from pg.types to set custom type parsers
+types.setTypeParser(types.builtins.DATE, (value) => value);
+types.setTypeParser(types.builtins.TIMESTAMP, (value) => value);
+types.setTypeParser(types.builtins.TIMESTAMPTZ, (value) => value);
+
+const ajv = new Ajv2020({ allErrors: true, strict: false });
+
+describe('pg_types', () => {
+    test('Add to Pagila', async () => {
+        await client.connect();
+
+        const tableName = 'all_data_types';
+
+        try {
+            await createTable();
+
+            await insertData();
+
+            // Generate schema in `introspect` package
+            const { schema } = await generate({
+                connectionString:
+                    'postgresql://postgres:postgres@localhost:5432/postgres',
+                includeSchemas: ['public'],
+                defaultSchema: 'public',
+                includeTables: [tableName],
+                outDir: '../introspect/src/tests/generated',
+            });
+
+            for (const [name, value] of Object.entries(await selectData())) {
+                const columnType = (schema.properties?.[tableName] as any)
```

Comment:
This definitely needs a bit of a refactor: 

Giving a good name to this functionality makes the intention clearer.

Essentially what you want is to check that a row matches the schema, let's create one:

```
function checkRowSchema(schema:Schema, table:string, row:unknown): void {}
```